src/decl.cpp

```diff
@@ -143,32 +143,32 @@ const Type *DeclSpecs::GetBaseType(SourcePos pos) const {
     if (soaWidth > 0) {
         const StructType *st = CastType<StructType>(retType);
 
-        if (st == NULL) {
+        if (st == nullptr) {
             Error(pos,
                   "Illegal to provide soa<%d> qualifier with non-struct "
                   "type \"%s\".",
-                  soaWidth, retType ? retType->GetString().c_str() : "NULL");
-            return NULL;
+                  soaWidth, retType ? retType->GetString().c_str() : "nullptr");
```

Comment:
Here we should have NULL as a ISPC language specifier

Reply:
It doesn't refer to ISPC NULL, but rather a placeholder for a missing `retType`, which might be the case when recovering from an error. But I agree that we should not change error messages.

Reply:
I've updated the PR and summarized what I did in "update" section of initial description.